lib/puppet/provider/service/systemd.rb

```diff
@@ -46,11 +67,22 @@ def status
   end
 
   def enable
+    output = systemctl("unmask", @resource[:name])
```

Comment:
Leaving a service masked and trying to start it is not a logical path. Either you want it masked and stopped or started. Masked and started is not a valid case.

So, we can either throw an error and ask the user what they actually want to do, or we can just unmask it. Either is fine with me.